### ISSUE TITLE
reg.create_key: ValueError: Could not convert the data to the specified type

### ISSUE THREAD

**Issue by xsteadfastx** [saltstack/salt#13916]
i try to get the example from the docs working. im running 2014.1.5 on the master and on the minion. the master is a ubuntu-server box and the client windows 7 ultimate. here is what i tried on the master:

```
salt 'tsmp2014060035' reg.create_key HKEY_CURRENT_USER 'SOFTWARE\Salt' 'version' '0.97'    
tsmp2014060035:
    The minion function caused an exception: Traceback (most recent call last):
      File "salt/minion.py", line 796, in _thread_return
      File "salt/modules/reg.py", line 142, in create_key
    ValueError: Could not convert the data to the specified type.
```

on the minion running the debugging mode:
```
[DEBUG   ] Command details {'tgt_type': 'glob', 'jid': '20140702124542103101', '
tgt': 'tsmp2014060035', 'ret': '', 'user': 'sudo_administrator', 'arg': ['HKEY_C
URRENT_USER', 'SOFTWARE\\Salt', 'version', 0.97], 'fun': 'reg.create_key'}
[WARNING ] The minion function caused an exception
Traceback (most recent call last):
  File "salt/minion.py", line 796, in _thread_return
  File "salt/modules/reg.py", line 142, in create_key
ValueError: Could not convert the data to the specified type.
[INFO    ] Returning information for job: 20140702124542103101
[INFO    ] User sudo_administrator Executing command reg.create_key with jid 201
40702124619831199
[DEBUG   ] Command details {'tgt_type': 'glob', 'jid': '20140702124619831199', '
tgt': 'tsmp2014060035', 'ret': '', 'user': 'sudo_administrator', 'arg': ['HKEY_C
URRENT_USER', 'SOFTWARE\\Salt', 'version', 0.97], 'fun': 'reg.create_key'}
[INFO    ] Returning information for job: 20140702124619831199
[INFO    ] User sudo_administrator Executing command reg.create_key with jid 201
40702124724589256
[DEBUG   ] Command details {'tgt_type': 'glob', 'jid': '20140702124724589256', '
tgt': 'tsmp2014060035', 'ret': '', 'user': 'sudo_administrator', 'arg': ['HKEY_C
URRENT_USER', 'SOFTWARE\\Salt', 'version', 0.97], 'fun': 'reg.create_key'}
[INFO    ] Returning information for job: 20140702124724589256
```

for the rest im not having problems with the windows minion. i mostly use it to install packages and stuff. thanks :)

**Comment by dariusjs** [comment]
Hi,

I also ran into this error report when using state modules. I found that if the value was not enclosed in single quotes then I would get the exact same report as the poster.

Below are my regedits for SSC Serv I was using and the error I got prior to enclosing values in single quotes:

----------
          ID: HKEY_LOCAL_MACHINE\SOFTWARE\octo\SSC Serv\Graphite\{2EAC0048-A701-4AA6-A160-1293B0CDCD55}\StoreRates
    Function: reg.present
      Result: False
     Comment: An exception occurred in this state: Traceback (most recent call last):
                File "salt/state.py", line 1529, in call
                File "salt/states/reg.py", line 65, in present
                File "salt/modules/reg.py", line 117, in set_key
              ValueError: Could not convert the data to the specified type.
     Started: 16:22:25.112000
    Duration: 16.0 ms
     Changes:


HKEY_LOCAL_MACHINE\SOFTWARE\octo\SSC Serv\Graphite\Enabled:
  reg.present:
    - value: 'True'
    - vtype: REG_SZ

HKEY_LOCAL_MACHINE\SOFTWARE\octo\SSC Serv\Graphite\{2EAC0048-A701-4AA6-A160-1293B0CDCD55}\Node:
  reg.present:
    - value: 'fiohb-graphite.corp.vattenfall.com'
    - vtype: REG_SZ

HKEY_LOCAL_MACHINE\SOFTWARE\octo\SSC Serv\Graphite\{2EAC0048-A701-4AA6-A160-1293B0CDCD55}\Service:
  reg.present:
    - value: '2003'
    - vtype: REG_SZ

HKEY_LOCAL_MACHINE\SOFTWARE\octo\SSC Serv\Graphite\{2EAC0048-A701-4AA6-A160-1293B0CDCD55}\StoreRates:
  reg.present:
    - value: 'True'
    - vtype: REG_SZ

HKEY_LOCAL_MACHINE\SOFTWARE\octo\SSC Serv\Graphite\{2EAC0048-A701-4AA6-A160-1293B0CDCD55}\Prefix:
  reg.present:
    - value: 'ssc_serv.'
    - vtype: REG_SZ

**Comment by twangboy** [comment]
The `reg` module was fixed in `2015.5`. These functions work properly in `2015.5` and `2015.8`.
